projects/cds/src/merchandising/facade/cds-merchandising-product.service.ts

```diff
@@ -1,49 +1,51 @@
 import { Injectable } from '@angular/core';
-import {
-  BaseSiteService,
-  LanguageService,
-  RoutingService,
-} from '@spartacus/core';
 import { combineLatest, Observable } from 'rxjs';
-import { map, switchMap } from 'rxjs/operators';
+import { flatMap, map } from 'rxjs/operators';
+import { MerchandisingProducts } from '../model/merchandising-products.model';
+import { MerchandisingUserContext } from '../model/merchandising-user-context.model';
 import { StrategyRequest } from './../../cds-models/cds-strategy-request.model';
 import { MerchandisingStrategyConnector } from './../connectors/strategy/merchandising-strategy.connector';
-import { MerchandisingProducts } from './../model/merchandising.products.model';
+import { MerchandisingSiteContext } from './../model/merchandising-site-context.model';
+import { CdsMerchandisingSiteContextService } from './cds-merchandising-site-context.service';
+import { CdsMerchandisingUserContextService } from './cds-merchandising-user-context.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CdsMerchandisingProductService {
   constructor(
     protected strategyConnector: MerchandisingStrategyConnector,
-    protected baseSiteService: BaseSiteService,
-    protected languageService: LanguageService,
-    private routingService: RoutingService
+    protected merchandisingUserContextService: CdsMerchandisingUserContextService,
+    protected merchandisingSiteContextService: CdsMerchandisingSiteContextService
   ) {}
 
   loadProductsForStrategy(
     strategyId: string,
     numberToDisplay?: number
   ): Observable<MerchandisingProducts> {
     return combineLatest([
-      this.baseSiteService.getActive(),
-      this.languageService.getActive(),
-      this.routingService
-        .getRouterState()
-        .pipe(map(state => state.state.params['productCode'])),
+      this.merchandisingSiteContextService.getSiteContext(),
+      this.merchandisingUserContextService.getUserContext(),
     ]).pipe(
-      map(([site, language, productId]: [string, string, string]) => {
-        const strategyRequest: StrategyRequest = {
-          site,
-          language,
-          pageSize: numberToDisplay,
-          productId,
-        };
-        return strategyRequest;
-      }),
-      switchMap(context =>
-        this.strategyConnector.loadProductsForStrategy(strategyId, context)
-      )
+      map(
+        ([siteContext, userContext]: [
+          MerchandisingSiteContext,
+          MerchandisingUserContext
+        ]) => {
+          const strategyRequest: StrategyRequest = {
+            ...siteContext,
+            ...userContext,
+            pageSize: numberToDisplay,
+          };
+          return strategyRequest;
+        }
+      ),
+      flatMap(context => {
```

Comment:
As `flatMap` is an alias for `mergeMap`, let's use `mergeMap` instead.

Reply:
Swapped for `mergeMap`